lib/Driver/Compilation.cpp

```diff
@@ -1365,7 +1365,14 @@ namespace driver {
       size_t NumTasks = TQ->getNumberOfParallelTasks();
       size_t NumFiles = PendingExecution.size();
       size_t SizeLimit = Comp.getBatchSizeLimit().getValueOr(DefaultSizeLimit);
-      return std::max(NumTasks, NumFiles / SizeLimit);
+
+      // An explanation of why the partition calculation isn't simple division.
+      // Using the default limit as an example, a module of 26 files must be
+      // compiled in 2 batches. Simple division yields 26/25 = 1 batch, but a
+      // single batch of 26 would exceed the limit of 25. To round up, the
+      // calculation is: `(x - 1) / y + 1`.
+      size_t NumPartitions = (NumFiles - 1) / SizeLimit + 1;
```

Comment:
Two comments:
1. Suggest you define a lambda for the round-up, or better yet, use a library function. Either way would make things a big clearer. (Your comment is great--keep it!)
2. I don't think (x - 1) / y + 1 is the canonical formula. The one I recall is (x + y - 1) / y. Your formula is equivalent, but it's too easy to read it as "(x - 1) / (y + 1)", which is how I read it. But best would be to just use existing library routines.

Reply:
Thanks I'll look to see if there's an existing function. I too considered `(x + y - 1) / y`, and for whatever reason I ended up this way. I'm happy to change it.

Reply:
Your forumla also handles x = 0.

Reply:
My pleasure. Good point!